src/internal_modules/roc_address/protocol_map.cpp

```diff
@@ -129,5 +129,44 @@ void ProtocolMap::add_proto_(const ProtocolAttrs& proto) {
     protos_[proto.protocol] = proto;
 }
 
+bool ProtocolMap::get_supported_interfaces(core::Array<Interface>& interface_array) {
```

Comment:
Nitpick: please move definitions of get_supported_interfaces and get_supported_protocols before definition of add_proto_, because usually we place definitions in .cpp in same order as declarations in header.